### ISSUE TITLE
Faster Nyx reading

### ISSUE THREAD

**Issue by andreufont** [igmhub/LaCE#24]
There must be a way to speed up the reading of the Nyx archive (and maybe also the Cabayol23 archive...).

We might be able to write the archive in a way that makes it faster to read, at least for the "standard archives" (Pedersen21, Cabayol23, Nyx). 

It is a bit frustrating to wait several minutes just to read the file again.

**Comment by andreufont** [comment]
Not a priority, though, but something to keep in mind

**Comment by jchavesmontero** [comment]
Laura and I had a similar problem when computing simulation parameters (using CAMB). Maybe it is faster for us because the code uses all cores available. The solution was precomputing everything we need it and storing this data into a file

**Comment by andreufont** [comment]
Is the cpu time dominated by calling CAMB? If we do a call per sim, it shouldn't be much (there only 15). 

Anyway, I agree that the solution is to store all we need in a new file once and for all.

**Comment by jchavesmontero** [comment]
We do one call per P1D measurement instead of per simulation. I will take a look into this once I come back from Poland

**Comment by andreufont** [comment]
Ouch. Well, that should be easy to change then. We can take a look at it once you are back.

**Comment by jchavesmontero** [comment]
I found that the time for reading Nyx goes down to 15 s if CAMB is not called at all. As significant fraction of the time goes into computing the conversion between km/s and Mpc.

**Comment by jchavesmontero** [comment]
By doing the conversion just once for all redshifts I managed to reduce the time by a factor of two. Reading all redshifts takes now

CPU times: user 52.6 s, sys: 736 ms, total: 53.3 s
Wall time: 20.8 s

**Comment by jchavesmontero** [comment]
The only difference was calling the function camb_cosmo.dkms_dMpc once per simulation instead of once per simulation and per snapshot (you can pass an array of z)

**Comment by andreufont** [comment]
Thanks for checkign. This problably means that we are doing something innecessary within dkms_dMpc (we should recycle the call to CAMB) but let's don't worry about this now (we do something similar in cup1d). The current timing is already pretty decent, and the notebooks are now dominated by training the Nyx emulator.

**Comment by jchavesmontero** [comment]
Ok! This is the commit that solves this issue in case you want to take a look 
https://github.com/igmhub/LaCE/commit/c63a3aa6df966297f650e8b514a575a3daedbc74

**Comment by andreufont** [comment]
Related to this #33

**Comment by andreufont** [comment]
A first speed up was implemented in #33

**Comment by andreufont** [comment]
Sorry, I didn't want to close the issue. 

A first speedup was implemented, and Jonás is working on another speedup to call dkms_dMpc once per simulation.

In parallel, we are also working on reading the cosmo params from file, to avoid CAMB calls (#33)

**Comment by andreufont** [comment]
This is being implemented in PR https://github.com/igmhub/LaCE/pull/29
